.eslintrc.js

```diff
@@ -139,11 +142,14 @@ module.exports = {
 				],
 				'@typescript-eslint/consistent-type-imports': ['error', { prefer: 'type-imports' }],
 
+				// disable consistent indexed object style. both are used in mvom
```

Comment:
We use `Record` and the index signature. The `Record` is good in most scenarios as used in `mvom` but,
TypeScript will report errors about recursive type reference in scenarios like in `Schema.ts` where `SchemaDefinition` references `SchemaTypeDefinition` and `SchemaTypeDefinition` references `SchemaDefinition`.

I think turning the rule off is better than trying to force one or the other. I think I liked the use of `Record` where it made sense and the use the index signature where needed as well.

Reply:
I would prefer to see a global enforcement of consistency and an override of the rule where it's not possible to use that. It looks like its by design that `Record` cannot circularly reference itself: https://github.com/microsoft/TypeScript/pull/33050#issuecomment-714348057

However, for those scenarios where that's not the case, it would be better to have consistency and enforce the rule.

Reply:
Good idea.

Reply:
Removed the override to the turn the rule off. I updated where possible to use `Record` and provided an override for where not possible.